lib/executor/ramping_arrival_rate.go

```diff
@@ -439,12 +460,13 @@ func (varr RampingArrivalRate) Run(parentCtx context.Context, out chan<- stats.S
 			}
 		}
 
-		select {
-		case vu := <-activeVUs: // ideally, we get the VU from the buffer without any issues
-			go runIteration(vu) //TODO: refactor so we dont spin up a goroutine for each iteration
+		// If iterations chan has a buffered message then
+		// there aren't any active VUs free to take care of a new request.
+		if len(iterations) == 0 {
+			iterations <- struct{}{}
```

Comment:
Hmm this doesn't seem right... Imagine if `len(iterations)` is `0`, but all VUs are currently busy doing some other iteration. Your code here will happily slot the new empty struct in `iterations`, but no VU will actually start executing it. Moreover, if you actually hadn't allocated all of the VUs up to `maxVUs` (i.e. if `remainingUnplannedVUs > 0`), you wouldn't start allocating a new VU here. So this iteration will be essentially stuck in limbo until a VU frees itself or a new iteration comes and we allocate a VU then.